Services/UserCourseStatusHistorizing/classes/class.ilUserCourseStatusHistorizingHelper.php

```diff
@@ -230,7 +230,13 @@ public function setIndividualStartAndEnd($user_id, $course_id, &$payload) {
 
 	public static function hasCertificate($user, $course)
 	{
+		require_once('./Services/Certificate/classes/class.ilCertificate.php');
+		require_once(__DIR__ . '/class.ilPrimitiveCertificateAdapter.php');
 		require_once './Modules/Course/classes/class.ilCourseCertificateAdapter.php';
+
+		$cert = new ilCertificate(new ilPrimitiveCertificateAdapter($course));
+		if($cert->readActive() == 0) { return 0; }
```

Comment:
`return false;` most probably. Lets not mix up booleans and integers, even if PHP does so.